qa/sc_dust_threshold_fork.py

```diff
@@ -0,0 +1,128 @@
+        )
+        sc_node2_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node.hostname, websocket_port_by_mc_node_index(0))),
+            submitter_private_keys_indexes=list(range(cert_max_keys))  # SC node owns all schnorr private keys.
+        )
+
+        network = SCNetworkConfiguration(SCCreationInfo(mc_node, 100, self.sc_withdrawal_epoch_length,
+                                                        cert_max_keys=cert_max_keys,
+                                                        cert_sig_threshold=cert_sig_threshold),
+                                         sc_node1_configuration, sc_node2_configuration)
+        self.sc_nodes_bootstrap_info = bootstrap_sidechain_nodes(self.options, network, 720 * 120 * 10)
+
+    def sc_setup_nodes(self):
+        return start_sc_nodes(2, self.options.tmpdir)
+
+    def run_test(self):
+        time.sleep(0.1)
+        self.sync_all()
+        mc_node = self.nodes[0]
+        sc_node = self.sc_nodes[0]
+        sc_node2 = self.sc_nodes[1]
+        sc_address_1 = sc_node.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        sc_address_2 = sc_node2.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        connect_sc_nodes(self.sc_nodes[0], 1)
+        self.sc_sync_all()
+        _50cent = 50
+
+        # Send some currency to sc_1
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      mc_node,
+                                      self.sc_nodes_bootstrap_info.genesis_account.publicKey,
+                                      self.sc_nodes_bootstrap_info.genesis_account_balance,
+                                      mc_node.getnewaddress())
+        mc_node.generate(1)
+        generate_next_block(sc_node, "first node")
+
+        # check we are still in pre-fork epoch
+        forging_info = http_block_forging_info(sc_node)
+        assert_equal(forging_info["bestEpochNumber"], 2)
+
+        # Send 50 satoshi to node2, it should be successful
+        sendCoinsToAddress(sc_node, sc_address_2, _50cent, fee=0)
+        generate_next_block(sc_node, "first node")
+        self.sc_sync_all()
+        check_wallet_coins_balance(sc_node2, 0.00000050)
+
+        # Create 2 more tx in mempools of sc1 and sc2 with value < 54 satoshi
+        sendCoinsToAddress(sc_node, sc_address_2, 0.00000037, fee=0)
+        sendCoinsToAddress(sc_node2, sc_address_1, 0.00000044, fee=0)
+
+        # switch to the next consensus epoch
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        forging_info = http_block_forging_info(sc_node)
+        assert_equal(forging_info["bestEpochNumber"], 3)
+
+        # assert that mempool transactions were rejected
+        self.sc_sync_all()
+        check_wallet_coins_balance(sc_node2, 0.00000050)
+        check_wallet_coins_balance(sc_node, 199.99999950)
```

Comment:
check the context of the generated block.
It actually contains both transactions generated.
* Why it occurs? Because the transaction was validated against the consensus epoch number before block apply, so even if this block supposes to activate the fork, fork rules are not checked here. It is related to the comment I left some time ago.
* Why the balance check pass the verification ? Because `sendCoinsToAddress` argument of amount must be specified in satoshi (as long, not decimal). If you will check the json representation of the transaction you will see ZenBox with value `0` (long(0.00000050) -> 0)

A few points not strictly related to current test are:
1) Why `check_wallet_coins_balance` works with decimal zen and `sendCoinsToAddress` with long satoshi ?
2) Why do we allow to create a ZenBox with `0` value? I thought it is impossible. But ok, it is appeared to be possible before fork activation, we allow any value from 0 to 21M of Zen